lsst-epo/investigations-client
# [C] refactor table of contents

JIRA: https://jira.lsstcorp.org/browse/EPO-8858

## What this change does ##

Refactor the Table of Contents to fix a few bugs, better match the designs, update localization, and use the new `Slideout` component

## Notes for reviewers ##

Client-side change only, you'll need to build and install the new libraries to test. 

Include an indication of how detailed a review you want on a 1-10 scale.
- 5

## Testing ##

- check that the regular slideout menu (on the left) is still good, it makes use of the new slideout too
- check that page numbers change on the table of contents button
- check that clicking the table of contents button moves focus to the "close" button in the table of contents and vice versa when closed
- check that section break and finish icons are in their correct places now
- check the logic for disabling pages in `useProgress`

## Checklist ##

If any of the following are true, please check them off

## Screenshots (optional) ##
### Before:
### After:
